### ISSUE TITLE
Function ghcnd is not working for some stations.

### ISSUE THREAD

**Issue by yuchenw** [ropensci/rnoaa#229]
```
R version 3.4.1 (2017-06-30)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows >= 8 x64 (build 9200)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] bindrcpp_0.2       dplyr_0.7.2        purrr_0.2.3        readr_1.1.1       
[5] tidyr_0.6.3        tibble_1.3.3       ggplot2_2.2.1.9000 tidyverse_1.1.1   
[9] rnoaa_0.7.0
```
I am trying to download weather data from a station. However, the `ghcnd` function is not working for some stations. For example,

```
library(tidyverse)
library(rnoaa)

stations <- ghcnd_stations()

stations %>% filter(id %in% "USC00332098")
```
```
# A tibble: 17 x 11
            id latitude longitude elevation state     name gsn_flag wmo_id element
         <chr>    <dbl>     <dbl>     <dbl> <chr>    <chr>    <chr>  <chr>   <chr>
 1 USC00332098  41.2783  -84.3847     213.4    OH DEFIANCE             HCN    TMAX
 2 USC00332098  41.2783  -84.3847     213.4    OH DEFIANCE             HCN    TMIN
 3 USC00332098  41.2783  -84.3847     213.4    OH DEFIANCE             HCN    TOBS
 4 USC00332098  41.2783  -84.3847     213.4    OH DEFIANCE             HCN    PRCP
 5 USC00332098  41.2783  -84.3847     213.4    OH DEFIANCE             HCN    SNOW
 6 USC00332098  41.2783  -84.3847     213.4    OH DEFIANCE             HCN    SNWD
 7 USC00332098  41.2783  -84.3847     213.4    OH DEFIANCE             HCN    DAPR
 8 USC00332098  41.2783  -84.3847     213.4    OH DEFIANCE             HCN    MDPR
 9 USC00332098  41.2783  -84.3847     213.4    OH DEFIANCE             HCN    WT01
10 USC00332098  41.2783  -84.3847     213.4    OH DEFIANCE             HCN    WT03
11 USC00332098  41.2783  -84.3847     213.4    OH DEFIANCE             HCN    WT04
12 USC00332098  41.2783  -84.3847     213.4    OH DEFIANCE             HCN    WT05
13 USC00332098  41.2783  -84.3847     213.4    OH DEFIANCE             HCN    WT08
14 USC00332098  41.2783  -84.3847     213.4    OH DEFIANCE             HCN    WT11
15 USC00332098  41.2783  -84.3847     213.4    OH DEFIANCE             HCN    WT14
16 USC00332098  41.2783  -84.3847     213.4    OH DEFIANCE             HCN    WT16
17 USC00332098  41.2783  -84.3847     213.4    OH DEFIANCE             HCN    WT18
# ... with 2 more variables: first_year <int>, last_year <int>
```
This shows that there are some station data in `USC00332098`. However, the `ghcnd` function could not download data.

`ghcnd(stationid = "USC00332098")`

> Error in scan(file = file, what = what, sep = sep, quote = quote, dec = dec,  : 
>   scan() expected 'an integer', got '"1893"'

`1893` seems to be the earliest year when the station collected data.

If I ran the example code from the help file, it works, indicating that at least the function can still work for some stations.
`ghcnd(stationid = "AGE00147704")`

```
# A tibble: 964 x 128
            id  year month element VALUE1 MFLAG1 QFLAG1 SFLAG1 VALUE2 MFLAG2 QFLAG2
         <chr> <int> <int>   <chr>  <int>  <chr>  <chr>  <chr>  <int>  <chr>  <chr>
 1 AGE00147704  1909    11    TMAX     NA                          NA              
 2 AGE00147704  1909    11    TMIN     NA                          NA              
 3 AGE00147704  1909    11    PRCP     NA                          NA              
 4 AGE00147704  1909    12    TMAX    170                    E    160              
 5 AGE00147704  1909    12    TMIN     90                    E    100              
 6 AGE00147704  1909    12    PRCP     24                    E      0 
...
```
But I cannot figure out why it fails for other stations.

**Comment by sckott** [comment]
thanks for your issue!

i just tried that example and it works  for me. I wonder if it was a temporary failure due to e.g., their network down. can you try it again and see if it still doesn't work

**Comment by yuchenw** [comment]
Thanks for looking into it. I just ran `ghcnd(stationid = "USC00332098")` and I still got the same error message. But `ghcnd(stationid = "AGE00147704")`  worked fine.

**Comment by sckott** [comment]
okay, run it again, but like `xx <- ghcnd(stationid = "USC00332098", config=httr::verbose())` to get verbose curl output - paste in the curl messages here 

and it it fails again, run `traceback()` and paste in output of that

**Comment by yuchenw** [comment]
Thanks. When I ran `xx <- ghcnd(stationid = "USC00332098", config=httr::verbose())`, I got the same error message. Running `traceback()` showed the output as follows: 

```
4: scan(file = file, what = what, sep = sep, quote = quote, dec = dec, 
       nmax = nrows, skip = 0, na.strings = na.strings, quiet = TRUE, 
       fill = fill, strip.white = strip.white, blank.lines.skip = blank.lines.skip, 
       multi.line = FALSE, comment.char = comment.char, allowEscapes = allowEscapes, 
       flush = flush, encoding = encoding, skipNul = skipNul)
3: read.table(file = file, header = header, sep = sep, quote = quote, 
       dec = dec, fill = fill, comment.char = comment.char, ...)
2: read.csv(csvpath, stringsAsFactors = FALSE, colClasses = ghcnd_col_classes)
1: ghcnd(stationid = "USC00332098", config = httr::verbose()) 
```

**Comment by sckott** [comment]
I’ll have to try this on a windows machine as it seems like another windows problem

**Comment by yuchenw** [comment]
Thanks for your time and help.  It may not be relevant, but I can get the weather data of this station using the `ncdc` function, although it would be great if the `ghcnd` function can work.

**Comment by yuchenw** [comment]
Today I had a chance to test the same code on a windows machine with `R3.3.2`. It works. Here are the session information.

```
R version 3.3.2 (2016-10-31)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] rnoaa_0.7.0

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.10     xml2_1.1.1       magrittr_1.5     rappdirs_0.3.1   munsell_0.4.3    colorspace_1.3-2 R6_2.2.0         stringr_1.2.0    httr_1.2.1      
[10] plyr_1.8.4       dplyr_0.5.0      hoardr_0.2.0     tools_3.3.2      grid_3.3.2       gtable_0.2.0     DBI_0.5-1        lazyeval_0.2.0   assertthat_0.1  
[19] digest_0.6.12    tibble_1.3.0     gridExtra_2.2.1  ggplot2_2.2.1    tidyr_0.6.3      curl_2.8.1       mime_0.5         stringi_1.1.3    scales_0.4.1    
[28] XML_3.98-1.5     jsonlite_1.3     lubridate_1.6.0 
```

**Comment by sckott** [comment]
@yuchenw good to know , thanks

**Comment by sckott** [comment]
@yuchenw first, it's a good idea to run the latest R distribution, we're currently on 3.4.1 

looking into the error still

**Comment by sckott** [comment]
I can't replicate this problem on 3.3.2 or 3.4.1 on windows - so closing this for now

**Comment by chanse-ford** [comment]
@sckott This is an old bug, but it seems like it never got figured out and for some reason I'm now having the same problem when I run ghcnd_search(). I've tried for multiple stations and get the same response every time. Session info is below.



` testdf <- ghcnd_search(TMaxStations[1])
Error in scan(file = file, what = what, sep = sep, quote = quote, dec = dec,  : 
  scan() expected 'an integer', got '"1887"'

sessionInfo()
R version 3.6.0 (2019-04-26)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 16299)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252    LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                           LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] rgdal_1.4-3     sp_1.3-1        rnoaa_0.8.4     isdparser_0.3.0 tibble_2.1.1    gridExtra_2.3   rappdirs_0.3.1  jsonlite_1.6   
 [9] xml2_1.2.0      XML_3.98-1.19   scales_1.0.0    ggplot2_3.1.1   tidyr_0.8.3     dplyr_0.8.1     lubridate_1.7.4 httr_1.4.0     

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.1       pillar_1.4.0     compiler_3.6.0   plyr_1.8.4       tools_3.6.0      digest_0.6.18    evaluate_0.13   
 [8] lattice_0.20-38  gtable_0.3.0     pkgconfig_2.0.2  rlang_0.3.4      rstudioapi_0.10  yaml_2.2.0       crul_0.7.4      
[15] curl_3.3         xfun_0.7         knitr_1.22       withr_2.1.2      stringr_1.4.0    grid_3.6.0       tidyselect_0.2.5
[22] glue_1.3.1       httpcode_0.2.0   R6_2.4.0         rmarkdown_1.12   hoardr_0.5.2     purrr_0.3.2      magrittr_1.5    
[29] htmltools_0.3.6  assertthat_0.2.1 colorspace_1.4-1 stringi_1.4.3    lazyeval_0.2.2   munsell_0.5.0    crayon_1.3.4 `

**Comment by sckott** [comment]
can you give a reproducible example please? i don't know what `TMaxStations[1]` is

**Comment by sckott** [comment]
@chanse-ford ^^

**Comment by chanse-ford** [comment]
Right. Sorry for not including it originally. Below is my code for running that.

`
library(rnoaa)
stations <- ghcnd_stations()

MIStations <- stations %>% 
  filter(state == "MI") %>%
  filter(first_year <= 2003)

MITMax <- MIStations %>% 
  filter(element == "TMAX")
TMaxStations <- as.vector(MITMax$id)


MITempdf <- MITMax[, 1:3]


TMaxdf <- as.data.frame(matrix(data = NA, ncol = 3))
colnames(TMaxdf) <- c("id", "tmax", "date")

for (i in 1:length(MITempdf$id)){
  templist <- ghcnd_search(MITempdf$id[i], var = "TMAX")
  tempdf <- as.data.frame(templist)
  tempdf <- tempdf[, 1:3]
  colnames(tempdf) <- c("id", "tmax", "date")
  TMaxdf <- merge.data.frame(TMaxdf, tempdf, by.x = c("id", "tmax", "date"), by.y = c("id", "tmax", "date"), all = TRUE, drop = FALSE)
}`

**Comment by sckott** [comment]
@chanse-ford  thanks, but give me the station that actually causes an error

**Comment by chanse-ford** [comment]
@sckott Sorry for the delay. That happens for any of the station codes I've tried. For instance, in the code I posted the for loop is supposed to run through each station in Michigan, but even other states I've pulled from within the Great Lakes throw out this error for some reason. I'll post a few examples below.


`
 >testdf <- ghcnd_search("USC00202250")
Error in scan(file = file, what = what, sep = sep, quote = quote, dec = dec,  : 
  scan() expected 'an integer', got '"1953"
> testdf <- ghcnd_search("USW00014836")
Error in scan(file = file, what = what, sep = sep, quote = quote, dec = dec,  : 
  scan() expected 'an integer', got '"1948"'
> testdf <- ghcnd_search("USC00475091")
Error in scan(file = file, what = what, sep = sep, quote = quote, dec = dec,  : 
  scan() expected 'an integer', got '"1919"'`



Interestingly it only seems to be US stations. When I ran the function on station "CA006158875" it seemed to work.

**Comment by sckott** [comment]
thanks

can you try installing the development version if you haven't already tried `remotes::install_github("ropensci/rnoaa")`. your session info above indicates you have the CRAN version. then try again

**Comment by chanse-ford** [comment]
@sckott I originally had the github version installed from trying to figure out the tibble issue with the ghcnd_stations() function a few weeks ago and started getting this error, so I reinstalled the CRAN version to see if it would fix it. Regardless I've reinstalled the github version and am still getting the error.

`
> testdf <- ghcnd_search("USC00202250")
Error in scan(file = file, what = what, sep = sep, quote = quote, dec = dec,  : 
  scan() expected 'an integer', got '"1953"'
> session_info()
Error in session_info() : could not find function "session_info"
> library(rnoaa)
> testdf <- ghcnd_search("USC00202250")
Error in scan(file = file, what = what, sep = sep, quote = quote, dec = dec,  : 
  scan() expected 'an integer', got '"1953"'
> sessionInfo()
R version 3.6.0 (2019-04-26)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 16299)
Matrix products: default
locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252   
attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base    
other attached packages:
[1] rnoaa_0.8.5.9421
loaded via a namespace (and not attached):
 [1] Rcpp_1.0.1       pillar_1.4.0     compiler_3.6.0   plyr_1.8.4       tools_3.6.0     
 [6] digest_0.6.18    jsonlite_1.6     lubridate_1.7.4  tibble_2.1.1     gtable_0.3.0    
[11] pkgconfig_2.0.2  rlang_0.3.4      rstudioapi_0.10  crul_0.7.4       curl_3.3        
[16] gridExtra_2.3    dplyr_0.8.1      stringr_1.4.0    xml2_1.2.0       rappdirs_0.3.1  
[21] grid_3.6.0       tidyselect_0.2.5 glue_1.3.1       httpcode_0.2.0   R6_2.4.0        
[26] XML_3.98-1.19    ggplot2_3.1.1    purrr_0.3.2      tidyr_0.8.3      hoardr_0.5.2    
[31] magrittr_1.5     scales_1.0.0     assertthat_0.2.1 colorspace_1.4-1 stringi_1.4.3   
[36] lazyeval_0.2.2   munsell_0.5.0    crayon_1.3.4    
`

**Comment by chanse-ford** [comment]
@sckott I think I've figured out the issue to my problem. I had originally used the function a year or so ago to download daily files. When I recently ran the code with a newer date range to incorporate the most recent data I didn't have refresh = TRUE, so it wasn't downloading the newer data sets since I still had the old data in my cache. When I set refresh to TRUE it seems to be working as normal.
